src/lib.rs

```diff
@@ -124,6 +124,9 @@ use std::boxed::Box;
 #[cfg(not(feature = "std"))]
 use alloc::boxed::Box;
 
+/// Error type that allows custom implementations of the Serialize trait.
+pub use ser::Error;
```

Comment:
I'm not sure. Having it in separate module kind of helps keeping the less-often used bits from polutting the more important ones (eg. in documentation).